src/main/kotlin/org/rust/ide/inspections/RsAssertEqualInspection.kt

```diff
@@ -71,12 +71,25 @@ class RsAssertEqualInspection : RsLocalInspectionTool() {
             }
         }
 
-        private fun buildAssert(project: Project, str: String, argPair: Pair<PsiElement, PsiElement>): PsiElement {
+        private fun buildAssert(project: Project, assertArgument: RsAssertMacroArgument): PsiElement? {
             val factory = RsPsiFactory(project)
-            val newAssert = factory.createExpression(str) as RsCallExpr
-            val args: MutableList<PsiElement> = mutableListOf(argPair.first)
-            args.add(factory.createComma())
-            args.add(argPair.second)
+            val newAssert = factory.createExpression("$assertName!()") as RsCallExpr
+
+            val argPair = comparedAssertArgs(assertArgument) ?: return null
+
+            val args: MutableList<PsiElement> = mutableListOf(
+                argPair.first,
+                factory.createComma(),
+                argPair.second
+            )
+
+            if (assertArgument.formatMacroArgList.isNotEmpty()) {
```

Comment:
I think we could remove this `if`, because the loop would be empty anyway.

Reply:
Yup, redundant check. 
1f7dd3a